src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -97,22 +87,28 @@ public class Builder : IDisposable
 					outdir = Path.Combine (XABuildPaths.TopDirectory, "bin", "Debug");
 				if (!Directory.Exists (Path.Combine (outdir, "lib")) || !File.Exists (Path.Combine (outdir, libmonodroidPath)))
 					outdir = Path.Combine (XABuildPaths.TopDirectory, "bin", "Release");
-				if (IsUnix) {
-					if (!Directory.Exists (Path.Combine (outdir, "lib")) || !File.Exists (Path.Combine (outdir, libmonodroidPath)))
-						outdir = "/Library/Frameworks/Xamarin.Android.framework/Versions/Current";
-					return Path.Combine (outdir, "lib", "xamarin.android");
-				}
-				else {
-					if (Directory.Exists (Path.Combine (outdir, "lib")) && File.Exists (Path.Combine (outdir, libmonodroidPath)))
-						return Path.Combine (outdir, "lib", "xamarin.android");
 
-					var visualStudioDirectory = TestEnvironment.GetVisualStudioDirectory ();
-					if (!string.IsNullOrEmpty (visualStudioDirectory))
-						return Path.Combine (visualStudioDirectory, "MSBuild", "Xamarin", "Android");
+				return outdir;
+			}
+		}
 
-					var x86 = Environment.GetFolderPath (Environment.SpecialFolder.ProgramFilesX86);
-					return Path.Combine (x86, "MSBuild", "Xamarin", "Android");
-				}
+		public string AndroidMSBuildDirectory {
+			get {
+				var msbuildDir = Path.Combine (BuildOutputDirectory, "lib", "xamarin.android", "xbuild", "Xamarin", "Android");
+				if (Directory.Exists (msbuildDir) && File.Exists (Path.Combine (msbuildDir, "lib", "armeabi-v7a", "libmono-android.release.so")))
+					return msbuildDir;
+
+				return TestEnvironment.MonoAndroidToolsDirectory;
+			}
+		}
+
+		public string FrameworkLibDirectory {
```

Comment:
Should we put a `///<summary>` comment on each of these properties to explain what path they point to? I will always be confused looking at these.

Something like:
```
FrameworkLibDirectory is the home of v1.0`, v10.0, etc., the path is:
Build tree - ./bin/Debug/lib/xamarin.android/...
madOS - /Library/Frameworks/Xamarin.Android.framework/...
Windows - C:\Program Files (x86)\Microsoft Visual Studio\...
```
Same for the other two properties here.

Reply:
👍